cEP-0016.md

```diff
@@ -0,0 +1,234 @@
+# Git Commit Content Inspection
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 16                                                |
+| Version  | 1.0                                               |
+| Title    | Git commit content inspection                     |
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+`GitCommitBear`.
+
+## Introduction
+
+The `GitCommitBear` at present performs a check on the content of regular
+commits made to coala. However, there are many special types of commit
+messages that should be used only in conjunction with patches containing a
+special type of content.
+
+For example, `git revert` commit which is used to revert any previous commit and
+`git merge` commit which is generated whenever someone performs a GitHub rebase
+and tries to merge master onto the pull request. Rebase and merge on GitHub will
+always update the committer information and create new commit SHAs which is not
+correct. These commits generate a default commit message that is very concise
+and not according to coala’s standards. The coala project wants to allow
+`git revert` commits, but does not want to accept `git merge` commits as the
+repositories are strictly fast-forward merges that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in `.coafile`
+
+## Background
+
+Below are some code snippets from `.coafile` of some projects that actively use
+`GitCommitBear`.
+
+```
+[commits]
+bears = `GitCommitBear`
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+shortlog_trailing_period = False
+```
+
+```
+[all.commit]
+bears = `GitCommitBear`
+shortlog_trailing_period = False
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+body_close_issue = True
+body_close_issue_full_url = True
+body_close_issue_on_last_line = True
+body_enforce_issue_reference = False
+```
+
+```
+[commit]
+bears = `GitCommitBear`
+```
+
+## Proposed Change
+
+This project would involve making three meta bears that would identify
+commits that skip CI buids, `git merge` and `git revert` commits.
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely Travis
+   merge commits, `git revert` , `git merge` and commits that disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For Travis commits:
```

Comment:
We need a consistent name for these things.  In point 1 above, a below, it is called a "Travis merge commit(s)".  Here you call it "Travis commit(s)".

Reply:
actually, *Travis merge commit* or *Travis commit* is not being introduced here without prior explanation. In the introduction, all I've read was about github merge commits. And on another project, what if I'm not using travis? I could be using semaphore, gitlab-ci, circle-ci, jenkins…

To me, the CI is not supposed to actually *create* commits, it's only supposed to approve of them. So I'd be happy to have some more context here ☺